core/profile.c

```diff
@@ -1032,34 +1039,34 @@ void calculate_deco_information(struct dive *dive, struct divecomputer *dc, stru
 			} else {
 				/* Keep updating the VPM-B gradients until the start of the ascent phase of the dive. */
 				if (decoMode() == VPMB && last_ceiling >= first_ceiling && first_iteration == true) {
-					nuclear_regeneration(t1);
-					vpmb_start_gradient();
+					nuclear_regeneration(ds, t1);
+					vpmb_start_gradient(ds);
 					/* For CVA iterations, calculate next gradient */
 					if (!first_iteration || in_planner())
-						vpmb_next_gradient(deco_state->deco_time, surface_pressure / 1000.0);
+						vpmb_next_gradient(ds, ds->deco_time, surface_pressure / 1000.0);
 				}
-				entry->ceiling = deco_allowed_depth(tissue_tolerance_calc(dive, depth_to_bar(entry->depth, dive)), surface_pressure, dive, !prefs.calcceiling3m);
+				entry->ceiling = deco_allowed_depth(tissue_tolerance_calc(ds, dive, depth_to_bar(entry->depth, dive)), surface_pressure, dive, !prefs.calcceiling3m);
 				if (prefs.calcceiling3m)
-					current_ceiling = deco_allowed_depth(tissue_tolerance_calc(dive, depth_to_bar(entry->depth, dive)), surface_pressure, dive, true);
+					current_ceiling = deco_allowed_depth(tissue_tolerance_calc(ds, dive, depth_to_bar(entry->depth, dive)), surface_pressure, dive, true);
 				else
 					current_ceiling = entry->ceiling;
 				last_ceiling = current_ceiling;
 				/* If using VPM-B, take first_ceiling_pressure as the deepest ceiling */
 				if (decoMode() == VPMB) {
 					if  (current_ceiling >= first_ceiling ||
-					     time_deep_ceiling == t0 && entry->depth == (entry - 1)->depth) {
+					     (time_deep_ceiling == t0 && entry->depth == (entry - 1)->depth)) {
```

Comment:
Just out of curiosity: Does this give a compiler warning w/o the parentheses?

Reply:
yes. clang does on Mac.